### PR TITLE
[couriositystream] Fix passing auth.

### PR DESCRIPTION
**Pull request by mnn**
### Description of your *pull request* and other information

</details>

Probably a week or two ago yt-dlp stopped downloading full videos, only downloaded "trial" ones (first 2 minutes). This happened also when I tried fresh master (commit 935bac1e).

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [ ] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

`flake8` passes, but I am not sure how to run specific tests. When running all tests it seemed there is a lot of missing videos from other extractors.

I have tested it on `https://curiositystream.com/video/5893`. Before the downloaded file was cut at 2 minutes (42MB), after this fix it has full duration of 11 minutes (229MB). Used command:

```
python -m yt_dlp --no-playlist --no-check-certificate --force-ipv4 --no-call-home -f 'bv+ba[language=en]' -v --cookies-from-browser vivaldi https://curiositystream.com/video/5893
```

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

**Comment by pukkandan** [comment]
> but I am not sure how to run specific tests. When running all tests it seemed there is a lot of missing videos from other extractors.

See point 6 - https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-support-for-a-new-site

File: yt_dlp/extractor/curiositystream.py [hunk 1]
```diff
@@ -23,9 +23,11 @@ def _call_api(self, path, video_id, query=None):
             auth_cookie = self._get_cookies('https://curiositystream.com').get('auth_token')
             if auth_cookie:
                 self.write_debug('Obtained auth_token cookie')
-                self._auth_token = auth_cookie.value
+                self._auth_token = compat_urllib_parse_unquote(auth_cookie.value)
+            else:
+                self.write_debug('Auth cookie NOT found')
```
Comment (pukkandan) [inline yt_dlp/extractor/curiositystream.py:28]: unnecessary

File: yt_dlp/extractor/curiositystream.py [hunk 2]
```diff
@@ -23,9 +23,11 @@ def _call_api(self, path, video_id, query=None):
...
         if self._auth_token:
-            headers['X-Auth-Token'] = self._auth_token
+            headers['x-auth-token'] = self._auth_token
```
Comment (pukkandan) [inline yt_dlp/extractor/curiositystream.py:30]: why the change?

**Comment by mnn** [comment]
> > but I am not sure how to run specific tests. When running all tests it seemed there is a lot of missing videos from other extractors.
> 
> See point 6 - https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-support-for-a-new-site

I tried it, but that doesn't pass on master already:

```
└─[1] <git:(master 935bac1e4✈) > python test/test_download.py TestDownload.test_CuriosityStream
[debug] Loaded 1731 extractors
[curiositystream] Extracting URL: http://app.curiositystream.com/video/2
[curiositystream] 2: Downloading JSON metadata
[curiositystream] 2: Downloading m3u8 information
[curiositystream] 2: Downloading JSON metadata
[curiositystream] 2: Downloading MPD manifest
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] 2: Downloading 1 format(s): hls-4426
[info] Writing video metadata as JSON to: test_CuriosityStream_2.info.json
F
======================================================================
FAIL: test_CuriosityStream (__main__.TestDownload):
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/dev/python/yt-dlp/test/test_download.py", line 215, in test_template
    expect_info_dict(self, tc_res_dict, tc.get('info_dict', {}))
  File "/mnt/dev/python/yt-dlp/test/helper.py", line 233, in expect_info_dict
    expect_dict(self, got_dict, expected_dict)
  File "/mnt/dev/python/yt-dlp/test/helper.py", line 186, in expect_dict
    expect_value(self, got, expected, info_field)
  File "/mnt/dev/python/yt-dlp/test/helper.py", line 178, in expect_value
    self.assertEqual(
AssertionError: 96.79 != 96.71 : Invalid value for field average_rating, expected 96.79, got 96.71

----------------------------------------------------------------------
Ran 1 test in 1.286s

FAILED (failures=1)
```

If I understand the test correctly, I doubt my change can affect it, since the test is not using auth at all.

File: yt_dlp/extractor/curiositystream.py (hunk 2 above)
Comment (mnn) [reply yt_dlp/extractor/curiositystream.py:30]: That's what official front-end is sending. If I remember correctly, headers' names *should* be case-insenstitive, but back-ends are not always adhering to it.

File: yt_dlp/extractor/curiositystream.py (hunk 1 above)
Comment (mnn) [reply yt_dlp/extractor/curiositystream.py]: It helped with debugging. This is not a first time this broke. Ok, I'll remove it. Personally, I would prefer some command line argument to tell yt-dlp to crash if auth fails. It is annoying start watching a video and then have it cut at 2 minutes, even though download was completed successfully.

**Comment by mnn** [comment]
I have updated the test. It seems fine, but please check it, since I don't really know (I am not the author of the extractor) why it started returning those extra fields...

File: yt_dlp/extractor/curiositystream.py (hunk 2 above)
Comment (pukkandan) [reply yt_dlp/extractor/curiositystream.py:28]: Our core code/urllib will not respect the case you set here, making this change unnecessary
```suggestion
            headers['X-Auth-Token'] = self._auth_token
```

File: yt_dlp/extractor/curiositystream.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import re
 
 from .common import InfoExtractor
-from ..compat import compat_str
+from ..compat import compat_str, compat_urllib_parse_unquote
```
Comment (pukkandan) [inline yt_dlp/extractor/curiositystream.py:4]: This is deprecated. `import urllib.parse` and use `urllib.parse.unquote`

File: yt_dlp/extractor/curiositystream.py (hunk 1 above)
Comment (pukkandan) [reply yt_dlp/extractor/curiositystream.py]: If login fails (`_perform_login`), error is raised, no?

Comment (pukkandan) [reply yt_dlp/extractor/curiositystream.py]: > It helped with debugging. 

The `if` block has a debug message, making this redundant

**Review by pukkandan — COMMENTED** [review]
> , since I don't really know (I am not the author of the extractor) why it started returning those extra fields...

Tests didn't always used to be this strict. So old tests are expected to fail with extra fields. You updating them is good

File: yt_dlp/extractor/curiositystream.py [hunk 4]
```diff
@@ -54,8 +54,11 @@ class CuriosityStreamIE(CuriosityStreamBaseIE):
             'description': 'Vint Cerf, Google\'s Chief Internet Evangelist, describes how he and Bob Kahn created the internet.',
             'channel': 'Curiosity Stream',
             'categories': ['Technology', 'Interview'],
-            'average_rating': 96.79,
+            'average_rating': 96.71,
             'series_id': '2',
+            'thumbnail': 'https://img.curiositystream.com/w:1255/cHJvZHVjdGlvbi9zb3VyY2VzL3RpdGxlcy8yL2hvcml6b250YWxfaW1hZ2VfMTZfOS9lbi1VUy83YjA2MWZjYS0yZDQyLTRiNGEtYTNkMy0zYmYzOGU0MGQzNzgvbWVkaWFfaG9yaXpvbnRhbF9pbWFnZV8xNl85XzJfMzg0MHgyMTYwXzAwMDEuanBn.jpg',
+            'tags': [],
+            'duration': 158
```
[the remark this answers is not in the document]
Comment (pukkandan) [reply yt_dlp/extractor/curiositystream.py]: pls verify my changes are correct

File: yt_dlp/extractor/curiositystream.py (hunk 1 above)
Comment (mnn) [reply yt_dlp/extractor/curiositystream.py]: > If login fails (`_perform_login`), error is raised, no?

It think it doesn't, when using cookies from browser for auth (it just doesn't send the header). But you are right, I should try using the auth arguments instead of taking it from the browser (it is more convenient to use cookies for me, because I am always logged-in in the browser; didn't have to complicate my custom download script to handle auth for different services).

File: yt_dlp/extractor/curiositystream.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (mnn) [reply yt_dlp/extractor/curiositystream.py]: They are working - test is still passing and the downloaded video is the full-length version :slightly_smiling_face:.
